src/create.cpp

```diff
@@ -158,42 +158,42 @@ static void displayCharacterHistory() {
     }
 }
 
-// Clear the previous history strings
-static void playerClearHistory() {
-    for (auto &entry : py.misc.history) {
-        entry[0] = '\0';
+static int16_t calculateSocialClass (int32_t social_class) {
```

Comment:
This whole function expects to be working with a int16_t so perhaps only accept that in the signature

Reply:
The problem is that randomNumber returns an int. I chose int32_t to ensure that the function returns 100 if social_class > 100. If there would be used int16_t, undetected overflows could occur. In this case the function would return a wrong value.